src/content/storage.js

```diff
@@ -0,0 +1,33 @@
+export default class Storage {
+  constructor() {
+    this._cache = {};
+    this._initialized = false;
+  }
+
+  init() {
+    if (this._initialized) {
+      return Promise.resolve();
+    }
+
+    this._initialized = true;
+    chrome.storage.onChanged.addListener(this._update);
+    return this._update();
+  }
+
+  get() {
+    return this._cache;
+  }
+
+  _update() {
+    return new Promise((resolve, reject) => {
+      chrome.storage.sync.get(data => {
+        if (chrome.runtime.lastError) {
+          reject(chrome.runtime.lastError);
+        }
+
+        this._cache = { ...data };
+        resolve();
+      });
+    });
+  }
+}
```

Comment:
I'm glad we're using Promises here, but it would be even nicer to use async/await. Chrome 55+ supports it natively, so we shouldn't need to change our Babel config.

Reply:
Do you have a specific pattern in mind? async/await requires a Promise, and so I think we have to wrap this call in a Promise.

Reply:
I see what you mean now, we're going to need the Promise constructor anyway so async/await wouldn't be useful in this file.

Reply:
I updated this to hopefully make it clearer! If we end up writing more wrapper functions around chrome storage methods we can move them into their own file.